### ISSUE TITLE
'int' object has no attribute 'name' error on my HA

### ISSUE THREAD

**Issue by Pastukhovar** [mill1000/midea-ac-py#32]
Just switched to your fork and can't get it working. Have in the logs:
```
сен 07 13:08:53 dacha-server docker[35217]: 2023-09-07 13:08:53.645 ERROR (MainThread) [homeassistant.components.climate] Error adding entities for domain climate with platform midea_ac
сен 07 13:08:53 dacha-server docker[35217]: Traceback (most recent call last):
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
сен 07 13:08:53 dacha-server docker[35217]:     await asyncio.gather(*tasks)
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 752, in _async_add_entity
сен 07 13:08:53 dacha-server docker[35217]:     await entity.add_to_platform_finish()
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1002, in add_to_platform_finish
сен 07 13:08:53 dacha-server docker[35217]:     self.async_write_ha_state()
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 730, in async_write_ha_state
сен 07 13:08:53 dacha-server docker[35217]:     self._async_write_ha_state()
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 830, in _async_write_ha_state
сен 07 13:08:53 dacha-server docker[35217]:     state, attr = self._async_generate_attributes()
сен 07 13:08:53 dacha-server docker[35217]:                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 773, in _async_generate_attributes
сен 07 13:08:53 dacha-server docker[35217]:     attr.update(self.state_attributes or {})
сен 07 13:08:53 dacha-server docker[35217]:                 ^^^^^^^^^^^^^^^^^^^^^
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 328, in state_attributes
сен 07 13:08:53 dacha-server docker[35217]:     data[ATTR_FAN_MODE] = self.fan_mode
сен 07 13:08:53 dacha-server docker[35217]:                           ^^^^^^^^^^^^^
сен 07 13:08:53 dacha-server docker[35217]:   File "/config/custom_components/midea_ac/climate.py", line 275, in fan_mode
сен 07 13:08:53 dacha-server docker[35217]:     return self._device.fan_speed.name.capitalize()
сен 07 13:08:53 dacha-server docker[35217]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
сен 07 13:08:53 dacha-server docker[35217]: AttributeError: 'int' object has no attribute 'name'
сен 07 13:08:53 dacha-server docker[35217]: 2023-09-07 13:08:53.649 ERROR (MainThread) [homeassistant.components.climate] Error while setting up midea_ac platform for climate
сен 07 13:08:53 dacha-server docker[35217]: Traceback (most recent call last):
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 367, in _async_setup_platform
сен 07 13:08:53 dacha-server docker[35217]:     await asyncio.gather(*pending)
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
сен 07 13:08:53 dacha-server docker[35217]:     await asyncio.gather(*tasks)
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 752, in _async_add_entity
сен 07 13:08:53 dacha-server docker[35217]:     await entity.add_to_platform_finish()
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1002, in add_to_platform_finish
сен 07 13:08:53 dacha-server docker[35217]:     self.async_write_ha_state()
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 730, in async_write_ha_state
сен 07 13:08:53 dacha-server docker[35217]:     self._async_write_ha_state()
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 830, in _async_write_ha_state
сен 07 13:08:53 dacha-server docker[35217]:     state, attr = self._async_generate_attributes()
сен 07 13:08:53 dacha-server docker[35217]:                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 773, in _async_generate_attributes
сен 07 13:08:53 dacha-server docker[35217]:     attr.update(self.state_attributes or {})
сен 07 13:08:53 dacha-server docker[35217]:                 ^^^^^^^^^^^^^^^^^^^^^
сен 07 13:08:53 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 328, in state_attributes
сен 07 13:08:53 dacha-server docker[35217]:     data[ATTR_FAN_MODE] = self.fan_mode
сен 07 13:08:53 dacha-server docker[35217]:                           ^^^^^^^^^^^^^
сен 07 13:08:53 dacha-server docker[35217]:   File "/config/custom_components/midea_ac/climate.py", line 275, in fan_mode
сен 07 13:08:53 dacha-server docker[35217]:     return self._device.fan_speed.name.capitalize()
сен 07 13:08:53 dacha-server docker[35217]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
сен 07 13:08:53 dacha-server docker[35217]: AttributeError: 'int' object has no attribute 'name'
```

**Comment by Pastukhovar** [comment]
Here is the logs
```
сен 08 11:47:25 dacha-server docker[35217]: 2023-09-08 11:47:25.159 INFO (MainThread) [custom_components.midea_ac] Starting midea-ac-py. Using msmart version 2023.9.2.
сен 08 11:47:25 dacha-server docker[35217]: 2023-09-08 11:47:25.159 INFO (MainThread) [custom_components.midea_ac] Querying device capabilities.
сен 08 11:47:26 dacha-server docker[35217]: 2023-09-08 11:47:26.984 INFO (MainThread) [custom_components.midea_ac.climate] Setting up climate platform.
сен 08 11:47:26 dacha-server docker[35217]: 2023-09-08 11:47:26.984 DEBUG (MainThread) [custom_components.midea_ac.climate] Supported operational modes: '[<HVACMode.FAN_ONLY: 'fan_only'>, <HVACMode.DRY: 'dry'>, <HVACMode.COOL: 'cool'>, <HVACMode.HEAT: 'heat'>, <HVACM>
сен 08 11:47:26 dacha-server docker[35217]: 2023-09-08 11:47:26.984 DEBUG (MainThread) [custom_components.midea_ac.climate] Supported preset modes: '['none', 'sleep', 'eco', 'boost']'.
сен 08 11:47:26 dacha-server docker[35217]: 2023-09-08 11:47:26.984 DEBUG (MainThread) [custom_components.midea_ac.climate] Supported fan modes: '['Auto', 'Full', 'High', 'Medium', 'Low', 'Silent']'.
сен 08 11:47:26 dacha-server docker[35217]: 2023-09-08 11:47:26.984 DEBUG (MainThread) [custom_components.midea_ac.climate] Supported swing modes: '['Off', 'Horizontal', 'Vertical', 'Both']'.
сен 08 11:47:26 dacha-server docker[35217]: 2023-09-08 11:47:26.986 INFO (MainThread) [custom_components.midea_ac.sensor] Setting up sensor platform.
сен 08 11:47:26 dacha-server docker[35217]: 2023-09-08 11:47:26.987 INFO (MainThread) [custom_components.midea_ac.binary_sensor] Setting up binary sensor platform.
сен 08 11:47:26 dacha-server docker[35217]: 2023-09-08 11:47:26.987 INFO (MainThread) [custom_components.midea_ac.switch] Setting up switch platform.
сен 08 11:47:27 dacha-server docker[35217]: 2023-09-08 11:47:27.317 ERROR (MainThread) [homeassistant.components.climate] Error adding entities for domain climate with platform midea_ac
сен 08 11:47:27 dacha-server docker[35217]: Traceback (most recent call last):
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
сен 08 11:47:27 dacha-server docker[35217]:     await asyncio.gather(*tasks)
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 752, in _async_add_entity
сен 08 11:47:27 dacha-server docker[35217]:     await entity.add_to_platform_finish()
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1002, in add_to_platform_finish
сен 08 11:47:27 dacha-server docker[35217]:     self.async_write_ha_state()
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 730, in async_write_ha_state
сен 08 11:47:27 dacha-server docker[35217]:     self._async_write_ha_state()
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 830, in _async_write_ha_state
сен 08 11:47:27 dacha-server docker[35217]:     state, attr = self._async_generate_attributes()
сен 08 11:47:27 dacha-server docker[35217]:                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 773, in _async_generate_attributes
сен 08 11:47:27 dacha-server docker[35217]:     attr.update(self.state_attributes or {})
сен 08 11:47:27 dacha-server docker[35217]:                 ^^^^^^^^^^^^^^^^^^^^^
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 328, in state_attributes
сен 08 11:47:27 dacha-server docker[35217]:     data[ATTR_FAN_MODE] = self.fan_mode
сен 08 11:47:27 dacha-server docker[35217]:                           ^^^^^^^^^^^^^
сен 08 11:47:27 dacha-server docker[35217]:   File "/config/custom_components/midea_ac/climate.py", line 275, in fan_mode
сен 08 11:47:27 dacha-server docker[35217]:     return self._device.fan_speed.name.capitalize()
сен 08 11:47:27 dacha-server docker[35217]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
сен 08 11:47:27 dacha-server docker[35217]: AttributeError: 'int' object has no attribute 'name'
сен 08 11:47:27 dacha-server docker[35217]: 2023-09-08 11:47:27.322 ERROR (MainThread) [homeassistant.components.climate] Error while setting up midea_ac platform for climate
сен 08 11:47:27 dacha-server docker[35217]: Traceback (most recent call last):
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 367, in _async_setup_platform
сен 08 11:47:27 dacha-server docker[35217]:     await asyncio.gather(*pending)
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
сен 08 11:47:27 dacha-server docker[35217]:     await asyncio.gather(*tasks)
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 752, in _async_add_entity
сен 08 11:47:27 dacha-server docker[35217]:     await entity.add_to_platform_finish()
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1002, in add_to_platform_finish
сен 08 11:47:27 dacha-server docker[35217]:     self.async_write_ha_state()
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 730, in async_write_ha_state
сен 08 11:47:27 dacha-server docker[35217]:     self._async_write_ha_state()
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 830, in _async_write_ha_state
сен 08 11:47:27 dacha-server docker[35217]:     state, attr = self._async_generate_attributes()
сен 08 11:47:27 dacha-server docker[35217]:                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 773, in _async_generate_attributes
сен 08 11:47:27 dacha-server docker[35217]:     attr.update(self.state_attributes or {})
сен 08 11:47:27 dacha-server docker[35217]:                 ^^^^^^^^^^^^^^^^^^^^^
сен 08 11:47:27 dacha-server docker[35217]:   File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 328, in state_attributes
сен 08 11:47:27 dacha-server docker[35217]:     data[ATTR_FAN_MODE] = self.fan_mode
сен 08 11:47:27 dacha-server docker[35217]:                           ^^^^^^^^^^^^^
сен 08 11:47:27 dacha-server docker[35217]:   File "/config/custom_components/midea_ac/climate.py", line 275, in fan_mode
сен 08 11:47:27 dacha-server docker[35217]:     return self._device.fan_speed.name.capitalize()
сен 08 11:47:27 dacha-server docker[35217]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I am using dockerized HA, so what going with msmart inside running container:

```
pip show msmart
WARNING: Package(s) not found: msmart
dacha-server:/config# pip show msmart-ng
Name: msmart-ng
Version: 2023.9.2
Summary: A Python library for local control of Midea (and associated brands) smart air conditioners.
Home-page: 
Author: 
Author-email: Tucker Kern <tuckkern@gmail.com>
License: MIT
Location: /usr/local/lib/python3.11/site-packages
Requires: httpx, pycryptodome
Required-by: 
```

**Comment by Pastukhovar** [comment]
Fixed it with `return self._device.fan_speed` on 275 line of `clipate.py`

**Comment by Pastukhovar** [comment]
Thanks!

**Comment by mill1000** [comment]
Released in 2023.9.3.
